ide-common/src/main/kotlin/org/digma/intellij/plugin/auth/AuthManager.kt

```diff
@@ -69,38 +70,102 @@ class AuthManager {
     //withAuth should not be called concurrently by multiple threads. it is called only from AnalyticsService.replaceClient.
     //AnalyticsService.replaceClient is called on startup and when a relevant settings change and is synchronized.
     //the analyticsProvider here will usually be a non proxied RestAnalyticsProvider
-    fun withAuth(analyticsProvider: RestAnalyticsProvider, url: String): AnalyticsProvider {
+    fun withAuth(analyticsProvider: RestAnalyticsProvider): AnalyticsProvider {
 
-        Log.log(logger::info, "wrapping analyticsProvider with auth for url={}", url)
+        Log.log(logger::info, "wrapping analyticsProvider with auth for url={}", analyticsProvider.apiUrl)
 
         //loginOrRefresh will fail if backend is not up. it should catch all possible exceptions so that a proxy will always be returned.
-        loginOrRefresh(analyticsProvider, url)
+        loginOrRefresh(analyticsProvider, analyticsProvider.apiUrl)
 
         //always return a proxy. even if login failed. the proxy will try to loginOrRefresh on AuthenticationException
-        return proxy(analyticsProvider, url)
+        return proxy(analyticsProvider, analyticsProvider.apiUrl)
     }
 
+    fun logout() {
+        val digmaAccount = DigmaDefaultAccountHolder.getInstance().account
 
-    private fun loginOrRefresh(analyticsProvider: RestAnalyticsProvider, url: String, forceRefresh: Boolean = false): Boolean {
+        digmaAccount?.let { account ->
+            runBlocking {
+                DigmaAccountManager.getInstance().removeAccount(account)
+                DigmaDefaultAccountHolder.getInstance().account = null
+            }
+        }
 
-        val result = loginOrRefreshInternal(analyticsProvider, url, forceRefresh)
+        fireChange()
+    }
 
-        val authInfo =  AuthInfo(DigmaDefaultAccountHolder.getInstance().account?.userId)
+    fun login(analyticsProvider: RestAnalyticsProvider, userName: String, password: String): LoginResult {
+        val loginResult = loginInternal(analyticsProvider, userName, password);
```

Comment:
login can be a public method that don't accept the analyticsProvider , see comment above about saving it as class member on withAuth